src/ctoken.ts

```diff
@@ -372,7 +409,15 @@ export function handleTransfer(event: Transfer): void {
       marketID,
     ])
   } else {
-    if (balanceOfFromAccountResult.value.equals(event.params.amount)) {
+    // We should only decrement supplierCount when the from balance is now zero -AND- the from is not the market contract
+    // We should not decrement supplier count when the from is the market contract (i.e. minting)
+    if ((balanceOfFromAccountResult.value.equals(zeroBI) && !event.params.from.equals(cTokenContract._address))) {
```

Comment:
When handling `Transfer` events, if the resulting balance of the `from` account is now 0 **AND** the `from` account is not the market contract (market contracts will also have a cToken balance of 0 unless someone has inadvertently sent some cTokens to it) we should decrement the `supplierCount` for that market.

The bug that caused negative supplier counts was due to the fact that `mint` calls initiate a `transfer` from the market contract to the minter. Because the `USDC.multi` market has a 0 `balanceOf` for its own cToken, this would cause mints to incorrectly decrement the `supplierCount`.

There is a unit test that mocks a cToken transfer from test account 1 to test account 2 and mocks the `balanceOf` contract calls. This type of transfer will now correctly decrement `supplierCount` **AND** increment `supplierCount` as both of these conditions are triggered on the `handleTransfer`.

Since all types of events that can result in cToken movement emit a `Transfer`, this should encompass all edge cases.